bpf/cilium-map-migrate.c

```diff
@@ -307,17 +307,18 @@ typedef int (*bpf_handle_state_t)(struct bpf_elf_ctx *ctx,
 				  const struct bpf_elf_map *map,
 				  const char *name, int exit);
 
-char fs_base[PATH_MAX + 1];
+char * fs_base = NULL;
 
 void fs_base_init()
 {
 	const char *mnt_env = getenv(BPF_ENV_MNT);
-
+	char file[PATH_MAX + 1];
 	if (mnt_env) {
-		snprintf(fs_base, sizeof(fs_base), "%s/tc/globals", mnt_env);
+		snprintf(file, sizeof(file) - 1, "%s/tc/globals", mnt_env);
 	} else {
-		strcpy(fs_base, "/sys/fs/bpf/tc/globals");
+		strcpy(file, "/sys/fs/bpf/tc/globals");
 	}
+	fs_base = strdup(file);
```

Comment:
Since this is just a synthetical case, could we just use `-Wnoformat-truncation` and let subsequent open request fail? After all the workaround here is not really fixing the warning either since you'd need 2*PATH_MAX +1 buffer which couldn't even be handled by path lookup anyway.

Reply:
all the file name variable are in PATH_MAX + 1 so subsequent open/stat/rename will fail.

Reply:
Right, which is fine, but the "only" thing this bit above does is to hide the fact from gcc that fs_base would be of size `PATH_MAX + 1`, since we now need to allocate it dynamically, hence, gcc cannot statically verify, and thus warn about it. Meaning, this bit is for the purpose of silencing the gcc warning in which case it would be much more straight forward to just add to CFLAGS disabling of this warning.

It also introduces two actual bugs: i) strdup() can fail but is not checked here (unlikely but not impossible), ii) we should also free(fs_base) in case fn == NULL test below.

Reply:
Thanks  @borkmann I updated the pr with Wno-format-truncation tested with gcc-7/8